openmrs-client/src/main/java/org/openmrs/mobile/api/retrofit/PatientApi.java

```diff
@@ -194,14 +194,21 @@ public void onFailure(Call<PatientPhoto> call, Throwable t) {
     public void updatePatient(final Patient patient, @Nullable final DefaultResponseCallbackListener callbackListener) {
         final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(OpenMRS.getInstance());
         Boolean syncstate = prefs.getBoolean("sync", true);
-        new PatientDAO().updatePatient(patient.getId(), patient);
         if (syncstate) {
             RestApi restApi = RestServiceBuilder.createService(RestApi.class);
             Call<Patient> call = restApi.updatePatient(patient, patient.getUuid(), "full");
             call.enqueue(new Callback<Patient>() {
                 @Override
                 public void onResponse(Call<Patient> call, Response<Patient> response) {
                     if (response.isSuccessful()) {
+                        patient.getPerson().setUuid(patient.getUuid());
```

Comment:
This is done here not before sending updated patient to server, because with the uuid person object would be changed in RequestBody to "person": "person_uuid" and this would not update the patient properly. We set it here to update patient photo.